DS and Algorithm/SeleniumTest/SearchResultsPage.java

```diff
@@ -0,0 +1,20 @@
+// src/main/java/SearchResultsPage.java
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class SearchResultsPage {
+    private WebDriver driver;
+
+    public SearchResultsPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public int getNumberOfResults() {
+        // Example: Extract the number of search results
+        WebElement resultsStats = driver.findElement(By.id("result-stats"));
+        String statsText = resultsStats.getText();
+        String[] statsArray = statsText.split(" ");
+        return Integer.parseInt(statsArray[1].replace(",", ""));
```

Comment:
**Suggestion**: Consider using WebDriverWait instead of a direct getText() call to ensure the element is present and avoid NoSuchElementException. This approach enhances reliability by explicitly waiting for conditions rather than assuming the element is immediately available. <br> **Code Suggestion**: 
 ```
 - WebElement resultsStats = driver.findElement(By.id("result-stats"));
 + WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
 + WebElement resultsStats = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("result-stats")));
 ```